packages/widgets-manager/src/manager.ts

```diff
@@ -1,51 +1,130 @@
-    iframe.id = `letstalk-app-${appId}`;
-    iframe.onload = appOnload;
-    
-    const urlParams = Object.keys(this.urlParams).map(key => `${key}=${encodeURIComponent(this.urlParams[key])}`).join('&');
-    const widgetAppUrl = `${this.baseUrl}/api/v1/widget_apps/${appId}?${urlParams}`;
-    fetch(widgetAppUrl).then((widgetAppResonse) => {
-      widgetAppResonse.json().then((appConfiguration: App) => {
-        if (appConfiguration.payload_type === 'html') {
-          iframe.src = appConfiguration.payload;
-        }
-        // Add css style tag with style rules
-        this.addStyleString(appConfiguration.settings.css);
-        
-        // Apply settings to the iframe style property
-        Object.keys(appConfiguration.settings.inlineCss).forEach((key: string) => {
-          iframe.style.setProperty(key, appConfiguration.settings.inlineCss[key]);
+  _createIframeForApp = (app: App, cell: GridCell) => {
+    if (!document.getElementById(`letstalk-app-${app.id}`)) {
+      const iframe = document.createElement('iframe');
+      iframe.id = `letstalk-app-${app.id}`;
+
+      if (app.payload_type === 'html') {
+        iframe.src = `${app.payload}?appId=${app.id}`;
+      }
+
+      // Apply settings to the iframe style property
+      Object.keys(app.settings.inlineCss).forEach((key: string) => {
+        iframe.style.setProperty(key, app.settings.inlineCss[key]);
+      });
+
+      try {
+        const positionStrategy = makePostionStrategy(app.settings.position.type);
+        const positionProps = positionStrategy.getPositionProps(app, cell);
+  
+        Object.keys(positionProps).forEach((key: string) => {
+          iframe.style.setProperty(key, positionProps[key]);
         });
-        // After setting all the configuration I append it to the dom
+  
         document.body.appendChild(iframe);
-      });
-    });
-  };
   
-  unMountApp = (appId: Number) => {
+        // Add css style tag with style rules
+        this._addStyleString(app);
+      } catch (error) {
+        console.error('Could not position app on the screen. Check your configuration', error)
+      }
+    }
+  }
+
+  _removeIframeForApp = (appId: number) => {
     const appIframe = document.getElementById(`letstalk-app-${appId}`);
+    const appStyles = document.getElementById(`letstalk-app-${appId}-styles`);
     if (appIframe) {
       document.body.removeChild(appIframe);
     }
+    if (appStyles) {
+      document.body.removeChild(appStyles);
+    }
+  }
+
+  _unMountApps = (apps: App[]): void => {
+    apps.forEach((app) => {
+      this.unMountApp(app.id);
+      this.gridManager.removeApp(app.id);
+    });
+  }
+
+  _mountApps = (cell: GridCell, apps: App[]) => {
+    apps.forEach((app) => {
+      this._createIframeForApp(app, cell);
+    });
+  }
+
+  mountApp = (appId: number) => {
+    this.fetchAppData(appId).then((widgetAppResonse) => {
+      widgetAppResonse.json().then((appConfiguration: App) => {
```

Comment:
red flag: nested thens of the same type of monad.

this should actually look like this:

```
.then((widgetAppResonse) => widgetAppResonse.json())
.then((appConfiguration: App) => {
```